x/config/decode.go

```diff
@@ -60,6 +60,14 @@ func (m attributeMap) Get(name string, dst interface{}) (ok bool, err error) {
 	return true, err
 }
 
+func (m attributeMap) keys() []string {
```

Comment:
nit: Even for private types, it's nice to differentiate between their API and their internal helper methods. I would call this `Keys()`.